src/request.rs

```diff
@@ -459,7 +476,33 @@ fn parse_url(url: URL) -> (bool, URL, Port, URL) {
                 }
                 _ => port.push(c),
             },
+            #[cfg(not(feature = "urlencoding"))]
             UrlParseStatus::Resource => resource.push(c),
+            #[cfg(feature = "urlencoding")]
+            UrlParseStatus::Resource => match c {
+                // All URL-'safe' characters, plus URL 'special characters' like &, #, =
```

Comment:
Want a quick sanity check on this match arm to match sure I'm covering everything we don't need to touch: anything that doesn't need encoding and [any special characters we're assuming we can leave alone](https://github.com/neonmoe/minreq/issues/67#issuecomment-1013468623)

Reply:
`?` and `/` are missing, but otherwise this seems good to me. Thought about if `%` should be included, but I think opting into the feature also implies that the user won't be doing any url encoding themselves. Maybe `%`'s getting encoded should be mentioned in the feature introduction?

Reply:
The user is required to encode certain characters (the 'special' ones, if they don't want them to be interpreted literally). For that reason I'll leave %s untouched